script.js

```diff
@@ -1,4 +1,339 @@
+    if (input !== "h" && input !== "s") {
+      return `Please input only hit or stand!<br>${outputMessage()}`;
+    } else if (input === "h") {
+      let playerMidGameAction = playerMidGameHitAction();
+      return playerMidGameAction;
+    } else if (input === "s") {
+      gameState = "computerMidgame";
+      return `You have chosen to stand at ${playerScore} points!<br><br> The Computer will now take it's turn!<br> ${outputMessageForCPU()}`;
+    }
+  } else if (gameState === "computerMidgame") {
+    let computerAction = checkComputerScoring();
+    return computerAction;
+  } else if (gameState === "result") {
+    gameState = "scoreboard";
+    let endGameResult = checkEndGameScenario();
+    return endGameResult;
+  } else if (gameState === "scoreboard") {
+    refreshEverything();
+    return `Scoreboard- You : Computer <br> ${playerWonTimes} : ${computerWonTimes}<br> Click Submit to start a new round!`;
+  }
+}
+
+// end of main
+
+// dealing initial hands small function.
+function dealInitialHands() {
+  for (let i = 0; i < 2; i++) {
+    playerHand.push(deck.pop());
+    computerHand.push(deck.pop());
+  }
+}
+
+//checking initial hand for blackjacks
+function checkInitialHands(playerHand, computerHand) {
+  let playerStartScore = 0;
+  let computerStartScore = 0;
+  let initialPhaseResult = "";
+  for (let i = 0; i < playerHand.length; i += 1) {
+    playerStartScore += playerHand[i].score;
+  }
+  for (let j = 0; j < computerHand.length; j += 1) {
+    computerStartScore += computerHand[j].score;
+  }
+  if (playerStartScore === computerStartScore && playerStartScore === 21) {
+    initialPhaseResult = "double blackjack";
+  } else if (playerStartScore === 21 && computerStartScore !== 21) {
+    initialPhaseResult = "player blackjack";
+  } else if (playerStartScore !== 21 && computerStartScore === 21) {
+    initialPhaseResult = "computer blackjack";
+  } else {
+    initialPhaseResult = "game continue";
+  }
+  return initialPhaseResult;
+}
+// function to reset the whole game
+function refreshEverything() {
+  gameState = "initialize";
+  deck = [];
+  deck = shuffleCards(makeDeck());
+  playerHand = [];
+  computerHand = [];
+  playerScore = 0;
+  computerScore = 0;
+  playerBustStatus = false;
+  computerBustStatus = false;
+}
+
+//function to display all players cards depending on current size of hand
+function callCards() {
+  let message = "";
+  for (i = 0; i < playerHand.length; i += 1) {
+    message += `<br>${playerHand[i].name} of ${playerHand[i].suit} [${playerHand[i].score}]`;
+  }
+  return message;
+}
+
+// function to display all cpu cards depending on current size of hand
+function callCPUCards() {
+  let message = "";
+  for (i = 0; i < computerHand.length; i += 1) {
+    message += `<br>${computerHand[i].name} of ${computerHand[i].suit} [${computerHand[i].score}]`;
+  }
+  return message;
+}
+
+//function to calculate player scores
+function calculatePlayerScore(playerHand) {
```

Comment:
this function and the one below is identical. Do you think you could combine them into a single function. You might want to relook into this concept because I think you might not fully understand functions yet.